meson.build

```diff
@@ -1880,3 +1880,28 @@ if get_option('cplayer')
     executable('mpv', sources, dependencies: dependencies, win_subsystem: 'windows,6.0',
                include_directories: includedir, install: true)
 endif
+
+output = [
+    '',
+    '',
+    '   mpv: ' + meson.project_version(),
+    '  ---------------------',
+    '',
+    '   Notable features',
+    '',
+    '       d3d11:       ' + d3d11.allowed().to_string(),
+    '       gpu-next:    ' + libplacebo_next.to_string(),
+    '       javascript:  ' + javascript.found().to_string(),
+    '       libmpv:      ' + get_option('libmpv').to_string(),
+    '       libplacebo:  ' + libplacebo.found().to_string(),
+    '       lua:         ' + lua['use'].to_string(),
+    '       macos:       ' + (cocoa.found() and swift.allowed()).to_string(),
+    '       opengl:      ' + (GL.found() or egl['use']).to_string(),
+    '       vulkan:      ' + vulkan.found().to_string(),
+    '       wayland:     ' + wayland['use'].to_string(),
+    '       win32:       ' + win32_desktop.to_string(),
+    '       x11:         ' + x11['use'].to_string(),
+    '',
+    '',
+]
+message('\n'.join(output))
```

Comment:
I would recommend instead using the `summary()` function, which is for modern meson i.e. versions which mpv requires, but gnome-remote-desktop almost certainly predated this when they originally implemented their hand-rolled messages.

The summary() function handles printing the project name+version, does spacing/alignment for you, can pretty-print true/false values using the green YES / red NO formatting used elsewhere (see the bool_yn kwarg), and can also be invoked from anywhere in meson.build -- or even multiple times. It will merge the messages from each invocation into the sections you specify, and emit all the logged summary statuses together at the end of the build (not at the time the summary() function is invoked).

Reply:
Oh, also, arbitrary messages are a bit obtuse in terms of what they do. Summaries have defined meaning -- they contain the final informational status of the build. So, muon (the c99 reimplementation of Meson) saves the summary into a plaintext file in the build directory, and has a dedicated command to print it again for you. `muon -C builddir/ info summary`. This is a neat idea I may implement in Meson too.

Obviously it relies on using that newfangled summary thing. :)

Reply:
Oh nice, next time I should more closely read the reference manual.